test/index.js

```diff
@@ -73,6 +79,12 @@ var squareRoot = function(n) {
                : S.Right(Math.sqrt(n));
 };
 
+//  highArity :: a -> b -> c -> d -> a
+var highArity = function(a, b, c, d) { return a; };
+
+//  thunk :: (a -> b) -> c -> a -> b
+var thunk = function(fn) { return function() { return fn; }; };
```

Comment:
The type signature can be more general:

```javascript
//  thunk :: a -> b -> a
var thunk = function(x) { return function(y) { return x; }; };
```

In fact, this is just [`S.K`][1], is it not?


[1]: http://sanctuary.js.org/#K